pre-processing/primod/ribamod/exchange_creator.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+from numpy.typing import NDArray
+from scipy.spatial import KDTree
+
+Int = np.int_
+Float = np.float_
+Bool = np.bool_
+
+
+def check_conductance(conductance: xr.DataArray) -> xr.DataArray:
+    # FUTURE: check that the conductance location does not change over time.
+    # The imod_coupler has static weights: changing locations means having
+    # to update the coupling weights over time as well.
+    if "time" in conductance.dims:
+        return conductance.isel(time=0, drop=True)
+    else:
+        return conductance
+
+
+def find_coupled_cells(
+    conductance: xr.DataArray,
+    gridded_basin: xr.DataArray,
+    basin_ids: "pd.Series[int]",
+) -> tuple[NDArray[Int], NDArray[Bool]]:
+    """
+    Compare the location of conductance with the gridded basin.
+
+    Either may have values where the other has NaN values.
+    Only cells that have values in the conductance AND in the gridded
+    basin should be coupled.
+    """
+    # Conductance is leading parameter to define location, for both river
+    # and drainage.
+    # FUTURE: check for time dimension? Also order and inclusion of layer
+    # in conductance.
+    # Use xarray where to force the dimension order of conductance.
```

Comment:
Maybe "Use xarray to find out where to force [..]"?

Reply:
Uh no, it's the xarray.where() function specifically, I'll clarify

Reply:
Ah 😅